components/Navbar/UserButton.tsx

```diff
@@ -35,9 +35,9 @@ export default function UserButton() {
               <button
                 onClick={() => handleClick(Modal.SignIn)}
                 aria-label="profile"
-                className="text-grey-300 h-10 cursor-pointer"
+                className="text-grey-300 flex h-full w-full cursor-pointer items-center justify-center hover:bg-black/25"
               >
-                <Icon icon="avatar" />
+                <Icon icon="avatar" className="h-[30px] w-[30px]" />
```

Comment:
If I read this correctly the button could now potentially be bigger than the icon (since button uses `h-full` and `w-full` whereas the icon is a fixed `30px`). I think its better if we make sure that the button is a fixed `30px` instead, and have the icon adapt its sized based on the button so we can ensure that wherever the user clicks on the icon, the event handler fires.

Reply:
This HTML structure is getting a bit out of hand for the user button. 

![image](https://github.com/saving-satoshi/saving-satoshi/assets/695901/9f82c1c3-d3de-40cc-b42c-e625253be2b7)

You should be able to do this without any `div` and `span`. Main issue I see is that there's a mix of flexbox and non-flexbox layout techniques. If you use flexbox, the container can define how children should resize so you can avoid all the `w-full` and `h-full`. I'd really recommend looking into making the markup more lean. The best flex box guide I've found is the one by [CSS-Tricks](https://css-tricks.com/snippets/css/a-guide-to-flexbox/), been pulling that one up 1,000 times. But let me know if there are good reasons for all this nesting.

Reply:
> If I read this correctly the button could now potentially be bigger than the icon (since button uses h-full and w-full whereas the icon is a fixed 30px).

I went for this approach due to the user button being used both on the homepage and the lesson pages where the space for the user icon is much larger. I keeping with the buttons on the navbar in the lesson pages i tried to also apply the interactivity to the full space of the icon in the navbar to act as the button